go/ocr2/decryptionplugin/decryption.go

```diff
@@ -269,11 +271,20 @@ func (dp *decryptionPlugin) Report(ctx context.Context, ts types.ReportTimestamp
 			continue
 		}
 
-		// OCR2.0 guaranties 2f+1 observations are from distinct oracles
-		// which guaranties f+1 valid observations and, hence, f+1 valid decryption shares.
-		// Therefore, here it is guaranteed that len(decrShares) > f.
+		// This is a sanity check.
+		// OCR2.0 guarantees 2f+1 observations are from distinct oracles.
+		// which guarantees f+1 valid observations and, hence, f+1 valid decryption shares.
+		// Therefore, here it should be guaranteed that len(decrShares) > f.
+		if len(decrShares) < fPlusOne {
+			dp.logger.Error("DecryptionReporting Report: not enough valid decryption shares after processing all observations, skipping aggregation of decryption shares", commontypes.LogFields{
```

Comment:
Here we are not calling decryptionQueue.SetResult() on purpose right?

Reply:
Yes, this is a plugin error, not a user related error. 
I prefer we use the decryptionQueue.SetResult() only for user related errors.